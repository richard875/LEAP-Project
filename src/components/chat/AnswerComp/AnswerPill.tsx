import Image from "next/image";
import { useTheme } from "next-themes";
import { useEffect, useState } from "react";

const AnswerPill = () => {
  const { systemTheme } = useTheme();
  const [mounted, setMounted] = useState(false);

  useEffect(() => {
    setMounted(true);
  }, []);

  if (!mounted) return null;

  return (
    <div className="w-fit mb-5 h-fit pl-2.5 pr-2.5 pt-2 pb-2 rounded-full border border-stone-300 dark:border-stone-500 flex items-center justify-center gap-2.5 select-none">
      {systemTheme === "light" ? (
        <Image
          className="select-none"
          src="/logo-small-light.png"
          alt="LawConnect Logo"
          width={20}
          height={20}
          priority
        />
      ) : (
        <Image
          className="select-none"
          src="/logo-small-dark.png"
          alt="LawConnect Logo"
          width={20}
          height={20}
          priority
        />
      )}
      <p className="text-sm font-medium">LawConnect</p>
    </div>
  );
};

export default AnswerPill;
