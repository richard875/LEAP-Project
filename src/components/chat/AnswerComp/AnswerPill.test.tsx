import { render, screen } from "@testing-library/react";

import "@testing-library/jest-dom";
import AnswerPill from "./AnswerPill";

describe("AnswerPill", () => {
  it("should render correctly", () => {
    const { container } = render(<AnswerPill />);
    expect(container).toBeTruthy();
  });

  it("should match the snapshot", () => {
    const { asFragment } = render(<AnswerPill />);
    expect(asFragment()).toMatchSnapshot();
  });

  it("should have the correct elements in the component", () => {
    render(<AnswerPill />);
    const logoLight = screen.getByAltText("LawConnect Logo");
    const companyName = screen.getByText("LawConnect");

    expect(logoLight).toBeInTheDocument();
    expect(companyName).toBeInTheDocument();
  });
});
