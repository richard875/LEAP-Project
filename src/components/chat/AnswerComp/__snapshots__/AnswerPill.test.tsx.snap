// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`AnswerPill should match the snapshot 1`] = `
<DocumentFragment>
  <div
    class="w-fit mb-5 h-fit pl-2.5 pr-2.5 pt-2 pb-2 rounded-full border border-stone-300 dark:border-stone-500 flex items-center justify-center gap-2.5 select-none"
  >
    <img
      alt="LawConnect Logo"
      class="select-none"
      data-nimg="1"
      decoding="async"
      height="20"
      src="/_next/image?url=%2Flogo-small-dark.png&w=48&q=75"
      srcset="/_next/image?url=%2Flogo-small-dark.png&w=32&q=75 1x, /_next/image?url=%2Flogo-small-dark.png&w=48&q=75 2x"
      style="color: transparent;"
      width="20"
    />
    <p
      class="text-sm font-medium"
    >
      LawConnect
    </p>
  </div>
</DocumentFragment>
`;
